src/include/transaction/transaction_manager.h

```diff
@@ -99,7 +99,7 @@ class TransactionManager {
   std::atomic<timestamp_t> time_{timestamp_t(0)};
 
   // TODO(Tianyu): This is the famed HyPer Latch. We will need to re-evaluate performance later.
-  common::SharedLatch commit_latch_;
+  common::Gate txn_gate_;
```

Comment:
We can remove the todo from Tianyu. Also should it be "commit_gate_" if that's the only place it's used (or do you foresee that changing?)

Reply:
I removed this comment and updated the comment in `TransactionManager::UpdatingCommitCriticalSection` to reflect the new state for the code.

With regard to the name of the gate, I think its a policy question of whether we name it for where the "gate" is (BeginTransaction) or where it gets locked/unlocked (Commit).  I'm fine either way, but it made more sense to me to label it by where the effect occurs.